test/widgets/chat/drive_chat_page_test.dart

```diff
@@ -0,0 +1,333 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:motis_mitfahr_app/account/models/profile.dart';
+import 'package:motis_mitfahr_app/account/widgets/avatar.dart';
+import 'package:motis_mitfahr_app/drives/models/drive.dart';
+import 'package:motis_mitfahr_app/drives/pages/drive_chat_page.dart';
+import 'package:motis_mitfahr_app/rides/models/ride.dart';
+import 'package:motis_mitfahr_app/util/chat/models/message.dart';
+import 'package:motis_mitfahr_app/util/chat/pages/chat_page.dart';
+import 'package:motis_mitfahr_app/util/supabase.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+import '../../util/factories/chat_factory.dart';
+import '../../util/factories/drive_factory.dart';
+import '../../util/factories/message_factory.dart';
+import '../../util/factories/model_factory.dart';
+import '../../util/factories/profile_factory.dart';
+import '../../util/factories/ride_factory.dart';
+import '../../util/mock_server.dart';
+import '../../util/pump_material.dart';
+import '../../util/request_processor.dart';
+import '../../util/request_processor.mocks.dart';
+
+void main() {
+  late Drive drive;
+  final MockRequestProcessor processor = MockRequestProcessor();
+  final Profile profile = ProfileFactory().generateFake(id: 1);
+
+  setUpAll(() {
+    MockServer.setProcessor(processor);
+  });
+
+  group('Has Stream Subscription', () {
+    setUp(() async {
+      whenRequest(processor).thenReturnJson([]);
+    });
+    testWidgets('when Drive has rides with active chat', (WidgetTester tester) async {
+      drive = DriveFactory().generateFake(
+        driverId: profile.id,
+        rides: [RideFactory().generateFake(chatId: 1, status: RideStatus.approved)],
```

Comment:
we should probably add another ride without chat here to have a more complete test case